pkg/cluster/kubeclient/roundtrip.go

```diff
@@ -167,8 +274,46 @@ func NewKubeclient(log *logrus.Entry, config *v1.Config, disableKeepAlives bool)
 	}
 
 	return &Kubeclientset{
-		Log:    log,
-		Client: cli,
-		Seccli: seccli,
+		Log:               log,
+		Client:            cli,
+		Seccli:            seccli,
+		disableKeepAlives: disableKeepAlives,
+		restconfig:        restconfig,
+		testConfig:        testConfig,
 	}, nil
 }
+
+// ReloadKubeClient will override dialers to call
+// API server via PrivateEndpoint while keeping certificate trust
+func (u *Kubeclientset) ReloadKubeClient(cs *api.OpenShiftManagedCluster) error {
```

Comment:
what is this for?

Reply:
the kubeclient is created earlier than when we have access to the privateEndpoint (returned from the deploy), so once we do know the privateEndpoint we have to re-create the kubeclients